### PR TITLE
Remove pop in AppBar hero animations

### PR DESCRIPTION
**Pull request by abarth**
Now the flexible space bar computes its effect from its size rather than from
the Scaffold's animation.

### CODE REVIEW

**Comment by abarth** [comment]
@HansMuller

File: packages/flutter/lib/src/material/flexible_space_bar.dart [hunk 1]
```diff
@@ -91,77 +71,63 @@ class _FlexibleSpaceBarState extends State<FlexibleSpaceBar> {
...
       final ThemeData theme = Theme.of(context);
-      final double fadeStart = (appBarHeight - toolBarHeight) / appBarHeight;
-      final double fadeEnd = (appBarHeight - toolBarHeight / 2.0) / appBarHeight;
-      final CurvedAnimation opacityCurve = new CurvedAnimation(
-        parent: _scaffoldAnimation,
-        curve: new Interval(fadeStart, fadeEnd)
-      );
-      final int alpha = new Tween<double>(begin: 255.0, end: 0.0).evaluate(opacityCurve).toInt();
+      final double opacity = (1.0 - (minHeight - currentHeight) / (kToolBarHeight - statusBarHeight)).clamp(0.0, 1.0);
+      final int alpha = (255.0 * opacity).round();
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/flexible_space_bar.dart]: Just use titleStyle.withOpacity() below?

**Comment by HansMuller** [comment]
LGTM

File: packages/flutter/lib/src/material/flexible_space_bar.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/material/flexible_space_bar.dart]: Good idea.

**Pull request merged**